File: kotlin/anagram/src/main/kotlin/Anagram.kt
class Anagram {
  var word: String

  constructor(word: String) {
    this.word = word.toLowerCase()
  }

  fun match(candidates: List<String>): Set<String> {
    return candidates.filter { isAnagram(it.toLowerCase())}
      .toSet()
  }

  private fun isAnagram(candidate: String): Boolean {
    return candidate != this.word &&
      candidate.toCharArray().sorted().joinToString("") == this.word.toCharArray().sorted().joinToString("")
  }
}

File: swift/anagram/Sources/Anagram/Anagram.swift
class Anagram {
    let word: String
    
    init(word: String) {
        self.word = word.lowercased()
    }
    
    func match(_ candidates: [String]) -> [String] {
        candidates.filter { isAnagram(candidate: $0.lowercased()) }
    }
    
    func isAnagram(candidate: String) -> Bool {
        candidate != word && candidate.sorted() == word.sorted()
    }
}
